lively.ide/studio/asset-browser.cp.js

```diff
@@ -0,0 +1,918 @@
+      return;
+    }
+
+    this.ui.searchClearButton.visible = true;
+
+    this.ui.assets.submorphs.forEach(s => {
+      if (!s.isAssetPreview) return;
+      if (!s.assetName.toLowerCase().includes(needle)) s.visible = s.isLayoutable = false;
+    });
+    if (this.ui.assets.submorphs.every(s => !s.isAssetPreview || !s.visible)) this.ui.noResultsIndicator.visible = true;
+    else this.ui.noResultsIndicator.visible = false;
+  }
+
+  confirm () {
+    // _promise is initialized inside of the popup, but not the window
+    if (this._promise) {
+      this._promise.resolve(this.selectedAsset.imageUrl);
+      this.container.remove();
+    } else {
+      const image = new Image({ imageUrl: this.selectedAsset.imageUrl });
+      image.determineNaturalExtent().then((extent) => {
+        const maxWidth = 500;
+        const maxHeight = 500;
+        const scaleFactor = Math.min(maxWidth / extent.x, maxHeight / extent.y);
+        image.extent = extent.scaleBy(scaleFactor);
+        image.openInWorld();
+      });
+    }
+  }
+
+  async initialize () {
+    this.ui.assetTypeSelector.enabled = false;
+    this.view.visible = false;
+    const li = $world.showLoadingIndicatorFor(null, 'Enumerating project assets...');
+    await this.listAssets();
+    this.ui.selectionButton.disable();
+    this.view.visible = true;
+    li.remove();
+  }
+
+  block () {
+    const fader = morph({
+      name: 'fader',
+      position: this.container.position,
+      extent: this.container.extent,
+      fill: Color.rgbHex('202020'),
+      opacity: 0.8
+    });
+    $world.addMorph(fader);
+    const li = $world.showLoadingIndicatorFor(this.container, 'Changing Assets in Properties Panel...');
+    once($world._assetBrowserPopup, 'close', () => {
+      fader.remove();
+      li.remove();
+      this.updateAtRuntime = true;
+      this.listAssets(true);
+    });
+  }
+
+  async activate () {
+    await this.initialize();
+    this._promise = promise.deferred();
+    return this._promise.promise;
+  }
+
+  async deleteAsset () {
+    const assetToDelete = resource($world.openedProject.url).join('assets/' + this.selectedAsset.fullFileName);
+    await assetToDelete.remove();
+    this.ui.selectionButton.disable();
+    this.selectedAsset.view.remove();
+    this.ui.deleteButton.visible = false;
+    this.selectedAsset = null;
+  }
+
+  selectAssetEntry (assetEntryModel) {
+    if (this.selectedAsset) this.selectedAsset.view.master.setState(this.view.getWindow() ? 'unselectedLight' : null);
+    this.selectedAsset = assetEntryModel;
+    this.ui.deleteButton.visible = true;
+    assetEntryModel.view.master.setState(this.assetBrowserAsPopup ? 'selectedDark' : 'selectedLight');
+    this.ui.selectionButton.enable();
+  }
+
+  async openFilePicker () {
+    const pickerOpts = {
+      multiple: true,
+      types: [
+        {
+          description: 'asset files',
+          accept: {
+            'image/*': ['.jpg', '.jpeg', '.gif', '.png', '.webp', '.jxl']
```

Comment:
It would be nice to have this info right next to the supported image formats regex defined in`lively.ide/assets.js`, right?